src/reyna/Reyna/Http/HttpClient.cs

```diff
@@ -12,11 +13,13 @@ public sealed class HttpClient : IHttpClient
     {
         public IConnectionManager ConnectionManager { get; set; }
         private IWebRequest webRequest;
+        private IReynaLogger Logger;
 
-        public HttpClient(IConnectionManager connectionManager, IWebRequest webRequest)
+        public HttpClient(IConnectionManager connectionManager, IWebRequest webRequest, IReynaLogger logger)
         {
             this.ConnectionManager = connectionManager;
             this.webRequest = webRequest;
+            Logger = logger;

```

Comment:
you might need to add a test that if the logger is null to throw NullArgumentException.
@Youhana-Hana